src/compiler/types.ts

```diff
@@ -2748,6 +2756,8 @@ namespace ts {
     export interface AnonymousType extends ObjectType {
         target?: AnonymousType;  // Instantiation target
         mapper?: TypeMapper;     // Instantiation mapper
+        elementType?: Type;      // Element expressions of evolving array type

```

Comment:
I don't follow. `autoArrayType` is the *declared* type of an auto-inferred `any[]` and it's already used outside of `getFlowTypeOfReference`.